docs(client): Update changesets and generate release notes

tylerbutler opened the pull request:
The release notes were generated using the command in https://github.com/microsoft/FluidFramework/pull/21951 including the minor metadata additions to the source changesets.

tylerbutler committed:
docs(client): Update changesets for clarity and concision

tylerbutler committed:
tinylicious-client

tylerbutler committed:
more

tylerbutler commented:
@microsoft/fluid-cr-docs FYI - this is still a draft but please take a look and leave feedback.

tylerbutler committed:
updates

Josmithr commented on .changeset/every-walls-grow.md:
```diff
@@ -3,6 +3,6 @@
 ---
 
-Detect arrayNode iterator invalidation
+fluid-framework, tree: Detect arrayNode iterator invalidation
```
I'm not a big fan of this pattern. These package names are already called out in the metadata above. Requiring us to call them out here again seems redundant.

This is aimed at denoting the "main" affected packages, right? Could we accomplish this by introducing syntax for the metadata at the top to capture this instead?

tylerbutler replied:
> I'm not a big fan of this pattern. These package names are already called out in the metadata above. Requiring us to call them out here again seems redundant.

I am not suggesting we manually do this moving forward. We do not have any code that creates these prefixes today, so it's all manual. If we like the layout, but not the implementation (manual), then we can add that to the list of features we want.

> This is aimed at denoting the "main" affected packages, right? Could we accomplish this by introducing syntax for the metadata at the top to capture this instead?

We could add more metadata, but my proposal is just to use the first package in the list as the "main" package. Again, we can automate stuff once we know what output we want.

Josmithr replied:
Why are we updating the title lines here then?

tylerbutler replied:
> Why are we updating the title lines here then?

Maybe I am confused... We are doing a release on Monday, and we won't have time to (a) learn what we want to automate and (b) automate it before that release. So for the release we'll still be using the existing release note tools, which take the changeset contents as-is.

Josmithr replied:
Oh, so these title updates are part of an existing automated flow? Sorry, I thought we were doing manual tinkering with the existing changelogs to support some new pattern.

tylerbutler replied:
Maybe the latest iteration will help clarify. The release notes are now included in this PR as they are produced by the tools we have. So you can see exactly what the automation produces and how the title changes make their way into the notes.

tylerbutler committed:
Merge branch 'main' into relnotes-2.1

tylerbutler committed:
updates

tylerbutler committed:
pnpm flub generate upcoming -g client -t minor --out RELEASE_NOTES/2.1.0.md

tylerbutler commented on RELEASE_NOTES/2.1.0.md:
```diff
@@ -0,0 +1,55 @@
+<!-- THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY. -->
+
+# Upcoming changes in Fluid Framework v2.1.0
```
Shouldn't say "upcoming changes"

tylerbutler commented on RELEASE_NOTES/2.1.0.md:
```diff
@@ -0,0 +1,55 @@
+
+# Upcoming changes in Fluid Framework v2.1.0
+
+## merge-tree: The Marker.fromJSONObject and TextSegment.fromJSONObject argument types have been corrected
```
Items should be grouped by package.

tylerbutler committed:
Merge branch 'main' into relnotes-2.1

alexvy86 commented on .changeset/gold-badgers-float.md:
```diff
@@ -2,21 +2,23 @@
 ---
 
-Added typeguard for SharedObjectKinds
+New feature: Added type guards for DDS types
```
This might be a challenge if we move to automatically adding package names as prefix to change titles. I like this better than having `shared-object-base` here.

Josmithr replied:
This sort of thing also seems like a good place to leverage conventional commits. I.e., derive this automatically from the PR CC tag (if present)

Josmithr replied:
Or just use a similar syntax in these files. Either works. I'm still a fan of CC for other reasons, but so long as we have a consistent pattern, I'm a happy camper 🙂

alexvy86 commented on .changeset/gold-badgers-float.md:
```diff
@@ -2,21 +2,23 @@
 }
 
-// With this change, that code can now be written like so:
+// In Fluid Framework 2.0 and beuyond, that code can now be written like so:
```
Typo
```suggestion
// In Fluid Framework 2.0 and beyond, that code can now be written like so:
```

alexvy86 reviewed (commented):
One typo to fix. Overall LGTM.

tylerbutler committed:
Update gold-badgers-float.md

Co-authored-by: Alex Villarreal <716334+alexvy86@users.noreply.github.com>

tylerbutler committed:
Merge branch 'main' into relnotes-2.1

tylerbutler committed:
Merge branch 'main' into relnotes-2.1

tylerbutler committed:
Merge branch 'main' into relnotes-2.1

tylerbutler committed:
updates

tylerbutler committed:
Port 2.0 release notes to main (#21941)

The 2.0 release notes were created in the release branch (#21600), so I
have cherry-picked them back into main.

CraigMacomber committed:
tree: Prefer type imports in VSCode workspace (#21942)

## Description

Tree's lints are configured to prefer type imports, so make
auto-importing match.

tylerbutler committed:
wip

tylerbutler committed:
updates

tylerbutler committed:
Merge branch 'main' into relnotes-2.1

tylerbutler committed:
update

jzaffiro commented on .changeset/gold-badgers-float.md:
````diff
@@ -2,21 +2,23 @@
 
 ```typescript
-// Works in FluidFramework 1.0 but not in the initial release of FluidFramework 2.0:
+// Works in Fluid Framework 1.0 but not in the initial release of FluidFramework 2.0:
````
nit: the spacing on `Fluid Framework` should be consistent
```suggestion
// Works in Fluid Framework 1.0 but not in the initial release of Fluid Framework 2.0:
```

jzaffiro commented on RELEASE_NOTES/2.1.0.md:
````diff
@@ -0,0 +1,131 @@
+There is now a drop-in replacement to `instanceof`: the static `.is()` method to `SharedObjectKind`, which is available on all DDSes. For example:
+
+```typescript
+// Works in Fluid Framework 1.0 but not in the initial release of FluidFramework 2.0:
````
same as above:
```suggestion
// Works in Fluid Framework 1.0 but not in the initial release of Fluid Framework 2.0:
```

tylerbutler committed:
updates

tylerbutler committed:
Merge branch 'main' into relnotes-2.1

tylerbutler committed:
regen notes

tylerbutler committed:
Merge branch 'main' into relnotes-2.1

tylerbutler committed:
Update .changeset/gold-badgers-float.md

Co-authored-by: jzaffiro <110866475+jzaffiro@users.noreply.github.com>

tylerbutler committed:
Update RELEASE_NOTES/2.1.0.md

Co-authored-by: jzaffiro <110866475+jzaffiro@users.noreply.github.com>

tylerbutler committed:
Merge branch 'main' into relnotes-2.1

tylerbutler commented:
If we like the resulting release notes in #21951 and the direction of the tool, we can operationalize it in a future release. It was already helpful this time even though it was still a semi-manual process. More notes in the other PR.

tylerbutler committed:
updates
